### DISCUSSION TITLE
New stride (video) tutorial ideas

### DISCUSSION THREAD

**Discussion by Aggror** [stride3d/stride#1780] [Ideas]
Hi everyone, 

I am looking into new ideas for (video) tutorials. Both with and without sample projects.

**Short tutorials**
These tutorials would stand on their own without perse being part of a group of videos or sample project.
- Font loading
- Community toolkit intro
- Fullscreen and other resolutions
- Lighting
- Post-processing/render pipeline customization
- Render targets 
- Camera layers
- Unit testing
- Physics constraints

**Tutorial sets**
This would a tutorial playlist covering a larger topic
- 2D
  - 2d Lighting/shadows
  - sprite animations
  - tilemaps
  - spritesheets
- Shaders and SDSL
  - Diffuse
  - Specular
  - Texturing (I guess there's a feature to expose shader properties to the property grid)
  - Using time, sin(time) etc. 
- 2d game?
  - Side scrolling: megaman style
  - Top down: Tank or Roguelike rpg
- 3d game?
  - Shooter? 
  - Platformer?
- Renewing editor tutorials
- Tutorials specifically for Unity users (Added by Vaclav, YouTube feedback)

**Comment by ultradevp** [comment]
how about a megamanx 3d side scroller its very simple and easy for both understanding the engine and for practice

**Comment by JoeAbiN** [comment]
Maybe make a series on rendering.
-Shaders and SDSL
-Lighting
-Post-processing/render pipeline customization

**Comment by Johnhersh** [comment]
I'm looking for anything 2D. I have a 2D game in Unity I'd love to port over. But I don't see anything in the docs about 2D at all. Especially lighting and shadows. But also sprite animations. Sprites. Tilemaps. All that stuff

**Comment by NicusorN5** [comment]
I suggest:
- shader tutorials
- how to use render targets, camera layers (etc)

**Reply by Aggror** [reply]
Can you elaborate on camera layers?

**Reply by Doprez** [reply]
My understanding on camera layers from the discord is they can be used for things like avoiding clipping one a FPS model when close to a wall. I would also love to  see some videos on SDSL in general,

**Comment by BlueGrassSquirrel** [comment]
basic game elements for people just starting game dev with Stride. Things like health/xp bars. Money systems. importing and setting up animations.  Timers and events.  
various other "welcome to game dev" things, but using Stride.   I feel like its another reason Unity gotten so popular. 
I remember the YouTube tuts back in Unity 4 getting me into development and solidifying my choice of Engine.

**Comment by Aggror** [comment]
Thanks, @BlueGrassSquirrel, @NicusorN5, @ultradevp , @Johnhersh , @JoeAbiN. 

I will update the topic with your suggestions.

**Comment by Wallaguiar** [comment]
Please make more simple videos in youtube, a 3D platform game for beginners, third person character, animations tree state machine controler in stride engine

**Comment by Wallaguiar** [comment]
The best way to popularize Stride Engine is through more YouTube videos

**Reply by GwinyaiBere** [reply]
we could have 5 YouTube channels covering different topics. Another way to popularize stride is by an asset store of some sort that provides an add-on marketplace.... That how the blender creates a means for blender users.

**Reply by Chimpchi** [reply]
Agreed, advertising is one thing, but if you want people to truly swap and stay, you need to show how much more convenient the engine is.

**Comment by GwinyaiBere** [comment]
Toon shader, cel shader, and basic animation blending.... also aim IK for a third person shooter aiming

**Comment by Chimpchi** [comment]
Animations, and audio would be nice to cover for the basics. Perhaps even showing how to trigger particles, audio, generic events, etc. through animation events.

**Reply by Aggror** [reply]
Animation and audio are already party of the intermediate video tutorials. 

Can you elaborate on 'triggering' something? Pressing a key? Walking intro a collider? Those last 2 are already covered.

**Comment by rkira** [comment]
please make tutorials for android too.
i recently faced a lot of issue of for setting up android workload on stride.

**Comment by MCVenor** [comment]
I have suggestions for the an "Experts" Series:
 - Using the GPU to create Voxel based Meshes. (This would include passing data to the GPU and back, in Unity this was done with the help of ComputeShaders, i dont know if stride has some equivalent thingy)
 - Using the Results of those to "layer" them together
 - Using both above to assign local space based textures

The total Result could be something like this:
![MeteorMcVenor](https://github.com/stride3d/stride/assets/12583072/e1614ef3-3bfe-4cb4-b604-84e0cae8fccb)

**Comment by niltor** [comment]
Terrain Tutorial: how to make terrains.

**Comment by IXLLEGACYIXL** [comment]
youtube short video on how to export blender with material so stride can find it
https://discord.com/channels/500285081265635328/740163288746426368/1245919830486548571
